tools/rest_api_reference_test.php

```diff
@@ -0,0 +1,349 @@
+    {
+        if (!is_file($restApiReference)) {
+            user_error("$restApiReference doesn't exist or is not a file", E_USER_ERROR);
+            exit(1);
+        }
+        if (!is_dir($dxpRoot)) {
+            user_error("$dxpRoot doesn't exist or is not a directory", E_USER_ERROR);
+            exit(2);
+        }
+
+        $this->restApiReference = $restApiReference;
+        $this->dxpRoot = $dxpRoot;
+        $this->parseApiReference($this->restApiReference);
+        $this->parseRoutes($consolePath, $routingFiles);
+    }
+
+    private function parseApiReference($restApiReference): void
+    {
+        $refRoutes = [];
+
+        $restApiRefDoc = new DOMDocument();
+        $restApiRefDoc->loadHTMLFile($restApiReference, LIBXML_NOERROR);
+        $restApiRefXpath = new DOMXpath($restApiRefDoc);
+
+        /** @var DOMElement $urlElement */
+        foreach ($restApiRefXpath->query('//*[@data-field="url"]') as $urlElement) {
+            if (!array_key_exists($urlElement->nodeValue, $refRoutes)) {
+                $refRoutes[$urlElement->nodeValue] = [
+                    'methods' => [],
+                ];
+            }
+            $refRoutes[$urlElement->nodeValue]['methods'][$urlElement->previousSibling->previousSibling->nodeValue] = true;
+        }
+
+        $this->refRoutes = $refRoutes;
+        ksort($this->refRoutes);
+    }
+
+    private function parseRoutes($consolePath = 'bin/console', $routingFiles = null)
+    {
+        if (is_string($consolePath)) {
+            $this->parseRouterOutput($consolePath);
+        } elseif (is_array($routingFiles)) {
+            $this->parseRoutingFiles($routingFiles);
+        } elseif (is_string($routingFiles)) {
+            $this->parseRoutingFiles([$routingFiles]);
+        } else {
+            $this->parseRoutingFiles([
+                'vendor/ibexa/rest/src/bundle/Resources/config/routing.yml',
+                'vendor/ibexa/commerce-rest/src/bundle/Resources/config/routing.yaml',
+                // `find $dxpRoot/vendor/ibexa -name "routing_rest.y*ml"`
+                //'vendor/ibexa/admin-ui/src/bundle/Resources/config/routing_rest.yaml',
+                'vendor/ibexa/calendar/src/bundle/Resources/config/routing_rest.yaml',
+                'vendor/ibexa/connector-dam/src/bundle/Resources/config/routing_rest.yaml',
+                'vendor/ibexa/personalization/src/bundle/Resources/config/routing_rest.yaml',
+                'vendor/ibexa/product-catalog/src/bundle/Resources/config/routing_rest.yaml',
+                //'vendor/ibexa/scheduler/src/bundle/Resources/config/routing_rest.yaml', // prefixed /api/datebasedpublisher/v1
+                'vendor/ibexa/taxonomy/src/bundle/Resources/config/routing_rest.yaml',
+            ]);
+        }
+        ksort($this->confRoutes);
+    }
+
+    private function parseRouterOutput($consolePath) {
+        $confRoutes = [];
+
+        $routerCommand = 'debug:router --format=txt';
```

Comment:
Why not `--format=json`?

Reply:
@Steveb-p To use `grep /api/ibexa/v2` directly on the command output instead of parsing the JSON then filtering on the `path` property of each route.